packages/core/src/storage.ts

```diff
@@ -208,6 +208,72 @@ export type FindResult<T extends Doc> = WithLookup<T>[] & {
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
 export interface TxResult {}
 
+/**
+ * @public
+ */
+export interface IndexedDoc {
+  id: Ref<Doc>
+  _class: Ref<Class<Doc>>
+  space: Ref<Space>
+  modifiedOn: Timestamp
+  modifiedBy: Ref<Account>
+  attachedTo?: Ref<Doc>
+  attachedToClass?: Ref<Class<Doc>>
+  searchTitle?: string
+  [key: string]: any
+}
+
+/**
+ * @public
+ */
+export interface SearchQuery {
+  query: string
+  classes?: Ref<Class<Doc>>[]
+  spaces?: Ref<Space>[]
+}
+
+/**
+ * @public
+ */
+export interface SearchOptions {
+  limit?: number
+}
+
+/**
+ * @public
+ */
+export interface SearchResultDoc {
+  id: Ref<Doc>
```

Comment:
I expected we will have same as for Inbox, like a record of parameters and a labels with IntlString, so UI will render them in proper places and will not decide about details.

So all after space should be ommited, except icon and probable color.

In this case no IndexedReaders and presenters will be required.

Thoughts?

Reply:
For notifications a presentation was relatively simple because it was a plain text. Now every component do something with fields before show them. Even name need to be processed before rendering. And there are avatar, issue's status which gives different icon and etc.

Now it's obvious which fields could be in SearchResultDoc. We definitely could replace this strict definition with JSON-like key-value structure, but what we could actually improve by that? I mean the way we retrieve value from IndexedDoc would be the same, and how we using it in component too. 

Or maybe do you mean we should change the way how we store data in index? But even then there would be an issue about what you need to know to show fields properly.